app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :post_owner?, only: %i[edit update destroy]
+
+  def index
+    @post = Post.new
+    @posts = Post.all
+  end
+
+  def show; end
+
+  def new
+    @post = Post.new
+  end
+
+  def edit; end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      # flash[:notice] = 'Post was successfully created.'
+      redirect_to @post
+    else
+      render :new
+    end
+  end
+
+  def update
+    render :edit unless @post.update(post_params)
+  end
+
+  def destroy
+    @post.destroy
```

Comment:
Upon deletion of a post the page you reroute to the `posts_path` so I don't have to reload the page manually to see the post deleted.